src/main/java/com/github/princesslana/eriscasper/action/GatewayAction.java

```diff
@@ -0,0 +1,26 @@
+package com.github.princesslana.eriscasper.action;
+
+import com.github.princesslana.eriscasper.gateway.OpCode;
+import io.reactivex.Completable;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface GatewayAction<I> extends Action {
```

Comment:
I'd actually prefer these to be abstract class, since execute is real functionality vs something computed from other fields.

I think semantically it's cleaner to keep real functionality out of interfaces

Reply:
I can make the abstract classes, yeah, sounds like a good idea